GDSCommon-Demo/GDSCommon-Demo/Mocks/Screens.swift

```diff
@@ -99,17 +99,19 @@ enum Screens: String, CaseIterable {
         case .gdsIconScreen:
             return IconScreenViewController()
         case .gdsQRCodeScanner:
-            let viewModel = MockQRScanningViewModel(dialogPresenter: dialogPresenter) { navigationController.popViewController(animated: true) } dismissAction: {}
+            let viewModel = MockQRScanningViewModel(dialogPresenter: dialogPresenter) { navigationController.popViewController(animated: true) } dismissAction: { }
             return ScanningViewController(viewModel: viewModel)
         case .gdsQRCodeScannerModal:
-            let viewModel = MockQRScanningViewModel(dialogPresenter: dialogPresenter) {  navigationController.dismiss(animated: true) } dismissAction: {}
+            let viewModel = MockQRScanningViewModel(dialogPresenter: dialogPresenter) { navigationController.dismiss(animated: true) } dismissAction: { }
             return ScanningViewController(viewModel: viewModel)
         case .gdsResultsView:
             return ResultsViewController(popToRoot: popToRoot, navController: navigationController)
         case .gdsResultsViewModal:
             return ResultsViewController(popToRoot: nil, navController: navigationController)
         case .gdsErrorView:
             return GDSErrorViewController(viewModel: MockErrorViewModel())
+        case .gdsErrorViewNoIcon:
+            return GDSErrorViewController(viewModel: MockErrorViewModelNoIcon())
         case .gdsErrorViewWithTertiary:
```

Comment:
When I run the demo app, the icon and tertiary error view appear as below :/ not sure why because the correct mock being used for each case, do you see this?

![simulator_screenshot_EAEA1C27-1E14-410D-8DAD-81382516CBC7](https://github.com/user-attachments/assets/a22dd6ed-451a-4501-a49f-60e01747982f)

Reply:
Good spot, the view controller was unwrapping the `viewModel` property which would be the `ErrorViewModelInitialiser` concrete type (only conforms to `GDSErrorViewModel`) if the concrete type doesn't conform to `GDSErrorViewModel`.
I've switched that to unwrap the `viewModelV2` and it works by passing a type conforming to either `GDSErrorViewModel` or `GDSErrorViewModelV2` & `GDSErrorViewModelWithImage` AND `GDSScreenWithTertiaryButtonViewModel` now 🙂

Reply:
Makes sense! thank you :)